contracts/simple_dex/lib.rs

```diff
@@ -155,6 +164,10 @@ mod simple_dex {
             amount_token_in: Balance,
             min_amount_token_out: Balance,
         ) -> Result<(), DexError> {
+            if self.halted {
```

Comment:
should we guard `deposit` and `withdrawal` as well? and some other write messages?